client/app.jsx

```diff
@@ -36,6 +38,19 @@ export default class App extends React.Component {
     this.setState({ user });
   }
 
+  handleSignOut() {
+    window.localStorage.removeItem('final-project-jwt');
+    this.setState({ user: null });
+    window.location.hash = 'sign-in';
+  }
+
+  handleProfileClick() {
+    const token = window.localStorage.getItem('final-project-jwt');
+    const payload = decodeToken(token);
+    const userId = payload.userId;
```

Comment:
If your user is already in `this.state`, you should be able to use that instead of pulling the token out of localstorage.

actually, is it possible for you to make that profile button an anchor with the right href and then remove this method?